src/features/envVisualizer/animationComponents/PopAnimation.tsx

```diff
@@ -0,0 +1,60 @@
+import { Easings } from 'konva/lib/Tween';
+import React from 'react';
+import { Group } from 'react-konva';
+
+import { ControlItemComponent } from '../compactComponents/ControlItemComponent';
+import { StashItemComponent } from '../compactComponents/StashItemComponent';
+import { Animatable, AnimatedTextboxComponent } from './AnimationComponents';
+import { getNodePositionFromItem } from './AnimationUtils';
+
+export class PopAnimation extends Animatable {
+  private popItemAnimation: AnimatedTextboxComponent;
+  private stashItemAnimation: AnimatedTextboxComponent;
+
+  constructor(popItem: ControlItemComponent, stashItem: StashItemComponent) {
+    super();
+    const popItemPosition = getNodePositionFromItem(popItem);
+    const stashItemPosition = getNodePositionFromItem(stashItem);
+    // TODO: improve the animation make the travel path an arc
+    this.popItemAnimation = new AnimatedTextboxComponent(
+      popItemPosition,
+      { ...stashItemPosition, opacity: 0 },
+      popItem.text
+    );
+    this.stashItemAnimation = new AnimatedTextboxComponent(
+      {
+        ...stashItemPosition,
+        x: stashItemPosition.x + stashItemPosition.width / 2,
+        offsetX: stashItemPosition.width / 2,
+        y: stashItemPosition.y + stashItemPosition.height / 2,
+        offsetY: stashItemPosition.height / 2
+      },
+      { scaleX: 0.6, scaleY: 0.6 },
+      stashItem.text,
+      { delayMultiplier: 0.3 }
```

Comment:
should we change this to 0.5, as discussed?

Reply:
this is the pop animation, i believe we discussed increasing the duration of the assignment and lookup animations instead (which are not part of this PR).